arangod/Scheduler/ThreadPoolScheduler.cpp

```diff
@@ -0,0 +1,132 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Business Source License 1.1 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+#include "ThreadPoolScheduler.h"
+
+#include <velocypack/Builder.h>
+#include <numeric>
+
+#include "Metrics/Gauge.h"
+#include "Metrics/Counter.h"
+
+using namespace arangodb;
+
+void ThreadPoolScheduler::toVelocyPack(velocypack::Builder& b) const {
+  b.add("scheduler-threads", VPackValue(0));  // numWorkers
+  b.add("blocked", VPackValue(0));            // obsolete
+  b.add("queued", VPackValue(0));             // scheduler queue length
+  b.add("in-progress",
+        VPackValue(0));                 // number of working (non-idle) threads
+  b.add("direct-exec", VPackValue(0));  // obsolete
+}
+Scheduler::QueueStatistics ThreadPoolScheduler::queueStatistics() const {
+  return QueueStatistics();
+}
+void ThreadPoolScheduler::trackCreateHandlerTask() noexcept {}
+void ThreadPoolScheduler::trackBeginOngoingLowPriorityTask() noexcept {
+  _metrics->_ongoingLowPriorityGauge += 1;
+}
+void ThreadPoolScheduler::trackEndOngoingLowPriorityTask() noexcept {
+  _metrics->_ongoingLowPriorityGauge -= 1;
+}
+void ThreadPoolScheduler::trackQueueTimeViolation() {}
+void ThreadPoolScheduler::trackQueueItemSize(std::int64_t) noexcept {}
```

Comment:
Just wondering why these two methods are not implemented here.

Reply:
Fixed